File: benchmarks/fixtures/webapp_csharp/Util/Logger.cs
namespace Webapp.Util;

using System;

/// <summary>
/// Simple structured logger for a named component.
/// </summary>
public class Logger
{
    private readonly string _name;

    public Logger(string name)
    {
        _name = name;
    }

    /// <summary>
    /// Factory method: creates a Logger for the given component name.
    /// </summary>
    public static Logger GetLogger(string name)
    {
        return new Logger(name);
    }

    public void Info(string message)
    {
        Console.WriteLine($"[INFO] [{_name}] {message}");
    }

    public void Warn(string message)
    {
        Console.WriteLine($"[WARN] [{_name}] {message}");
    }

    public void Error(string message)
    {
        Console.WriteLine($"[ERROR] [{_name}] {message}");
    }

    public void Debug(string message)
    {
        Console.WriteLine($"[DEBUG] [{_name}] {message}");
    }
}

File: benchmarks/fixtures/webapp_cpp/util/logger.cpp
#include "util/logger.h"

#include <iostream>
#include <string>
#include <utility>

namespace webapp {
namespace util {

Logger::Logger(std::string name) : name_(std::move(name)) {}

/// Factory: creates a Logger for the given component name.
Logger Logger::get_logger(const std::string& name) {
    return Logger(name);
}

void Logger::info(const std::string& message) const {
    std::cout << "[INFO] [" << name_ << "] " << message << std::endl;
}

void Logger::warn(const std::string& message) const {
    std::cout << "[WARN] [" << name_ << "] " << message << std::endl;
}

void Logger::error(const std::string& message) const {
    std::cout << "[ERROR] [" << name_ << "] " << message << std::endl;
}

void Logger::debug(const std::string& message) const {
    std::cout << "[DEBUG] [" << name_ << "] " << message << std::endl;
}

const std::string& Logger::name() const {
    return name_;
}

}  // namespace util
}  // namespace webapp
